### PR TITLE
Subclassing pd.DataFrame with the addition of units

### PR DESCRIPTION
**Pull request by GOLoDovkA-A**
Simple extension of the Pandas.DataFrame that keeps track of units for the dataframe.
Basic information about methods of classes extention you can find here:
https://docs.python.org/2/tutorial/classes.html#inheritance
Basic information about methods of extention in pandas you can find here:
https://pandas.pydata.org/docs/development/extending.html#extension-types
Suitable example:
https://github.com/pandas-dev/pandas/issues/10349
Much more suitable example:
https://tardis-sn.github.io/tardis/running/interaction/isotope_pd_subclass.html?highlight=isotopeabundances

## Description

Created class UDataFrame inherited from pd.DataFrame which contains metadata about units. Thats only shows the use of constructing new subclasses of pd.DataFrame.
## Motivation and Context

Pandas DataFrame isn't capable to store the units of the attributes. The other issue is that this information about units of attributes gets further vanished when we try to copy one dataframe into another. 

## How Has This Been Tested?

I've created an instances of UDataFrame copy them and checked the units.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

**Comment by wkerzendorf** [comment]
good! did you follow any guidelines for the pandas subclassing - can you add this in the header of the PR? Any good PR should have tests and documentation as well. This is the next step to add.

**Comment by GOLoDovkA-A** [comment]
Okay I'll do that now and also you said that we can chat about making a radioactive decay project.

**Comment by GOLoDovkA-A** [comment]
And actualy I didnt follow any guidlines exept https://tardis-sn.github.io/tardis/running/interaction/isotope_pd_subclass.html?highlight=isotopeabundances. I created that and just searching for anwers on my questions on stackoverfow

**Comment by GOLoDovkA-A** [comment]
Also I have lots of usefull links on this guidlines to extand my class

File: tardis/units/first_objective.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+        self.units = pd.Series(kwargs.pop("units", None))
+        super().__init__(*args, **kwargs)
+
+a = UDataFrame({"first_col": [1,2,3], "sec_col": [4, 5, 6]}, units=["cm", "mm"])
+
+b = a.copy()
+
+print(a, "Units for a:", a.units, " ",sep="\n"*2)
+
+print("Units for b:", b.units,sep="\n"*2)
```
Comment (marxwillia) [inline tardis/units/first_objective.py]: These lines should be part of a separate unit test.

**Comment by marxwillia** [comment]
Also please change the name of the file. First_objective is not descriptive of the functionality.

**Comment by GOLoDovkA-A** [comment]
Does next step should be about addition of simple slicing and sum of UDataFrame?

File: tardis/units/test.py [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+Test of class UDataFrame by making an instance and checking of units preservation after copying
+"""
+
+a = UDataFrame({"first_col": [1,2,3], "sec_col": [4, 5, 6]}, units=["cm", "mm"])
+
+b = a.copy()
+
+print(a, "Units for a:", a.units, " ",sep="\n"*2)
+
+print("Units for b:", b.units,sep="\n"*2)
```
Comment (marxwillia) [inline tardis/units/test.py]: Please look at the other tardis unit tests to see how to write a proper unit test using pytest.

**Comment by marxwillia** [comment]
@GOLoDovkA-A Are you planning on continuing with this PR?

**Comment by GOLoDovkA-A** [comment]
> @GOLoDovkA-A Are you planning on continuing with this PR?
Yes, im still working with tests and trying to add new features

**Pull request closed without merging**